tests/CsvFileTest.php

```diff
@@ -0,0 +1,356 @@
+<?php
+
+namespace Keboola\Csv\Tests;
+
+use Keboola\Csv\CsvFile;
+use PHPUnit\Framework\TestCase;
+
+class CsvFileTest extends TestCase
+{
+
+    public function testExistingFileShouldBeCreated()
+    {
+        self::assertInstanceOf(CsvFile::class, new CsvFile(__DIR__ . '/data/test-input.csv'));
+    }
+
+    /**
+     * @expectedException \Keboola\Csv\Exception
+     * @expectedExceptionMessage Cannot open file
+     */
+    public function testExceptionShouldBeThrownOnNotExistingFile()
+    {
+        $csv = new CsvFile(__DIR__ . '/something.csv');
```

Comment:
Je IMO lepší $this->expectException() a dát ho až na to nejposlednější místo, kde to má nastat (tady tady pod tím). Takže by to potenciálně prošlo, i kdyby vyhodil tu exception constructor. Speciálně v případě, kdy se vyhazuje nějaká "generická" exception.

Reply:
cool